core/src/test/scala/org/apache/spark/sql/delta/perf/OptimizeMetadataOnlyDeltaQuerySuite.scala

```diff
@@ -319,4 +321,16 @@ class OptimizeMetadataOnlyDeltaQuerySuite
       optimizationDisabledQueryPlan
     }
   }
+
+  test(".collect() and .show() both use this optimization") {
+    val collectPlans = DeltaTestUtils.withLogicalPlansCaptured(spark, optimizedPlan = true) {
+      spark.sql(s"SELECT COUNT(*) FROM $testTableName").collect()
```

Comment:
no way to verify results even for `collect`?

Reply:
AFAIK we can't verify the results of `.show()`.

For collect, sure we can verify the results.